Graph options - add VM and error states

rudyhuyn opened the pull request:
### Description of the changes:
- Add GraphSettingsViewModel to simplify the UI part of the control and manage all the logic
- Manage error states
- Disable keyboard when the flyout is opened
- Only load GraphSettings after the user clicks on the button to improve launch performance
- Keep default styling of textbox in GraphSettings
- Use TextBlock::Header instead of a TextBlock
- Add a 1px margin between radiobutton/togglebutton to improve the readability of the UI
- Tweak a little the graph buttons

### How changes were validated:
- Manually

Commit:
PR after merge and reset onto feature\graphingcalculator

Commit:
PR comment updates.

sanderl committed:
Updated GraphingInterfaces and added logic to enable key graph features for equations with variables

sanderl committed:
Removed LockAspectRatio, hooked up the engine for graph options and started fixing the styling on the page.

sanderl committed:
Updated UI for GraphSettings

sanderl committed:
Updated spacing, added graph options title, started the error state on the textboxes

rudyhuyn committed:
GraphControl: refactoring and optimizations (#831)

* GraphControl cleaning

* replace textbox value after submission

* rebase

* rebase

* rebase

* Add filters

* rebase!

* rebase

rudyhuyn committed:
Save and restore the value of EquationInputArea when users scroll (#866)

* Save and restore the value of EquationInputArea when users scroll

* clean up

* replace ->Focus(..) by FocusManager::TryFocus(...)

Commit:
add VM and manage error states

sanderl commented on src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.cpp:
```diff
@@ -0,0 +1,35 @@
+#include "pch.h"
```
Needs a copyright header.

sanderl commented on src/Calculator/GraphingSettingsViewModel.cpp:
```diff
@@ -0,0 +1,2 @@
+#include "pch.h"
```
do you still need this file?

rudyhuyn replied:
No

sanderl commented on src/Calculator/GraphingSettingsViewModel.h:
```diff
@@ -0,0 +1 @@
+#pragma once
```
Do you need this file?

sanderl commented on src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml.cpp:
```diff
@@ -4,12 +4,15 @@
 #include "CalcViewModel\Common\AppResourceProvider.cpp"
```
Copyright header

sanderl commented on src/Calculator/Views/GraphingCalculator/GraphingSettings.xaml.h:
```diff
@@ -6,7 +6,7 @@
 #pragma once
```
copyright header

sanderl reviewed (changes requested):
I did a buddy build and a buddy test. A few things I noticed:
1. For the error state, the background should change to red like the EquationTextBox
2. I noticed sometimes the grid doesn't update according to the options set by the user. 
I graphed sin(x), I noticed the grid options did not reflect what was shown on the graph (it showed -12 and 12 for x and -10 and 10 for y). I pressed enter to get the graph to reset and nothing happened. The graph showed the axes at -1000 to 1000. 

I like that you pulled most of the logic into a VM. I think that is the right design.

I made a few comments about extra files and missing headers.

Commit:
- Make GraphSettingsButton compatible with high contrast mode
- Fix crash when users open Graph Settings in high contrast mode
- Replace brushes created for the fake Toggle Button in Graph Settings by brushes used in the real ToggleButton

rudyhuyn commented:
Thank you for your feedback! Copyright added and extra file removed!

I've also fixed some issues with high contrast: ChromeWhiteHighControlSelectedForegroundBrush was missing in HighContrast and the toggle button in GraphControl had some issues.

I've also modified the RadioButton template to use the real ToggleButton brushes, so we will always be consistent with the system (and we won't impact app launch performance with extra brushes).

Should we move this PR of a PR (😂) to the main repository and do a PR instead?

Commit:
update VM

rudyhuyn committed:
Fix focus issues (#868)

rudyhuyn committed:
Update graph tooltip design and content (#867)

* update graph tooltip

* modify font size

rudyhuyn committed:
Remove EquationTextBox from KeyGraphFeaturesPanel (#833)

* remove EquationBox from KFG

* polish

* improve templates

* fix issue with dark theme and MathRichEditBox

Commit:
Set TrigUnitsRadioButtonStyle.FocusVisualMargin

Commit:
replace tabs

rudyhuyn committed:
support ctrl+- and + (#882)

Commit:
Prevent users to enter a non-double.
Force a redraw after X/Y Min/Max values are modified

Commit:
Update grid range when we reopen the popup

Commit:
modify renderMain

Commit:
rollback vcxproj

Commit:
prevent crash when users copy text

EriWong committed:
Wire up keyboard in graphing calculator (#863)

* Wire up graphing calculator keyboard with math rich edit control

* CR feedback

* Handle focus bug in flyout

Commit:
force std::stod to use the full string

Commit:
rebase

rudyhuyn committed:
High Contrast support for Graphing Calculator (#878)

* Equation Control - High contrast

* tweak

* Update graph colors when users switch from one high contrast mode to another

* decrease opacity of reveal borders

Commit:
remove keyboarding enabling/disabling

Commit:
merge with master

Commit:
- improve graph mock to support bitmap, graphingOption, graphRenderer
- calculate a proportional range in the ViewModel instead of relying on the graph

Commit:
Fix issue with zoom
